### PR TITLE
338-feat: Integrate visual regression into cicd

### LINKED ISSUE #338: Integrate Visual Regression Testing into CI/CD
## 📝 Describe the feature

Integrate Playwright with Github Actions

## ℹ️ Additional context

Add any other context about the feature here.

## ✅ Definition of Done (DoD):
-  A visual regression testing tool is integrated into the CI/CD pipeline.
-  Automated screenshots of key components/pages are generated during the CI/CD process.
-  Baseline images are correctly stored and accessible for comparison.
-  Visual differences are automatically detected and reported, with clear documentation on how to interpret the reports.
-  The CI/CD build fails if significant visual differences are detected, prompting a review by the developers.
-  Documentation is provided, including setup instructions, guidelines for updating baseline images, and troubleshooting tips.

### PR DESCRIPTION
**Pull request by SpaNb4**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description
Integrates Playwright Visual Testing with CI/CD using [Percy](https://percy.io/)

## Related Tickets & Documents

- Closes #338

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## Summary by CodeRabbit

- **New Features**
	- Introduced a visual testing workflow that can be triggered via comments on pull requests.
	- Added automated visual tests for key application pages, including the Community and Courses sections.

- **Bug Fixes**
	- Corrected naming inconsistencies in test scripts for Playwright.

- **Documentation**
	- Added a guide on implementing automated visual testing with Percy and Playwright.

- **Chores**
	- Updated configuration files to streamline testing processes and improve organization of test results.

### CODE REVIEW

**Comment by SpaNb4** [comment]
run visual now

File: readme/end-to-end-testing.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// TODO update docs
```
Comment (ansivgit) [inline readme/end-to-end-testing.md:1]: Why don't update docs now?

File: src/shared/__tests__/visual/community.spec.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import percyScreenshot from '@percy/playwright';
```
Comment (ansivgit) [inline src/shared/__tests__/visual/community.spec.ts:1]: As far as I understand, we decided to add all pages to the tests?

File: playwright.config.ts [hunk 3]
```diff
@@ -1,39 +1,23 @@
 import { defineConfig, devices } from '@playwright/test';
 
+// eslint-disable-next-line no-restricted-exports
```
Comment (Quiddlee) [inline playwright.config.ts:3]: footgun

File: src/shared/__tests__/visual/community.spec.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { test } from '@playwright/test';
+import { takeScreenshot } from './utils';
+
+test('Community page', async ({ page }) => {
+  await page.goto('/community');
```
Comment (Quiddlee) [inline src/shared/__tests__/visual/community.spec.ts:5]:
```suggestion
  await page.goto(ROUTES.COMMUNITY);
```

File: src/shared/__tests__/visual/courses.spec.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { test } from '@playwright/test';
+import { takeScreenshot } from './utils';
+
+test('Courses main page', async ({ page }) => {
+  await page.goto('/courses');
```
Comment (Quiddlee) [inline src/shared/__tests__/visual/courses.spec.ts:5]:
```suggestion
  await page.goto(ROUTES.COURSES);
```

File: src/shared/__tests__/visual/main.spec.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { test } from '@playwright/test';
+import { takeScreenshot } from './utils';
+
+test('Main page', async ({ page }) => {
+  await page.goto('/');
```
Comment (Quiddlee) [inline src/shared/__tests__/visual/main.spec.ts:5]:
```suggestion
  await page.goto(COURSES.HOME);
```

File: playwright.config.ts (hunk 3 above)
Comment (SpaNb4) [reply playwright.config.ts:3]: I can remove it, but then it'll be with that annoying yellow underline
image

File: src/shared/__tests__/visual/community.spec.ts (hunk 4 above)
Comment (SpaNb4) [reply src/shared/__tests__/visual/community.spec.ts]: Done, but it adds some new warnings 🙂
image

File: playwright.config.ts (hunk 3 above)
Comment (Quiddlee) [reply playwright.config.ts]: i mean our rule is a footgun for such case

File: src/shared/__tests__/visual/community.spec.ts (hunk 4 above)
Comment (Quiddlee) [reply src/shared/__tests__/visual/community.spec.ts]: we're going to refactor `app` folder soon and will move all the constants to `shared`

**Review by Quiddlee — APPROVED** [review]

**Review by ansivgit — APPROVED** [review]

**Review by KristiBo — APPROVED** [review]

**Pull request merged**